core-java-modules/core-java-nio-2/src/main/java/com/baeldung/lock/FileLocks.java

```diff
@@ -0,0 +1,239 @@
+		try (RandomAccessFile file = new RandomAccessFile(path.toFile(), "r"); // could also be "rw", but "r" is sufficient for reading
+				FileLock lock = file.getChannel().lock(from, size, true)) {
+			if (lock.isValid()) {
+				log.debug("This is a valid shared lock");
+				return lock;
+			}
+			return null;
+		} catch (Exception e) {
+			log.error(e.getMessage());
+		}
+		return null;
+	}
+
+	
+
+	static class Writer implements Runnable {
+
+		private Path path;
+
+		private volatile RandomAccessFile file;
+
+		private String text;
+
+		private volatile CountDownLatch countDownLatch;
+
+		/**
+		 * 
+		 * @param path           The path to the file we will write into
+		 * @param text           The text to write
+		 * @param countDownLatch A counter for thread synchronization
+		 * 
+		 */
+		public Writer(Path path, String text, CountDownLatch countDownLatch) {
+			this.path = path;
+			this.text = text;
+			this.countDownLatch = countDownLatch;
+		}
+
+		@Override
+		public void run() {
+			try {
+				lockAndWrite();
+			} catch (InterruptedException | FileNotFoundException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+			countDownLatch.countDown();
+		}
+
+		private void lockAndWrite() throws InterruptedException, FileNotFoundException {
+			ByteBuffer buffer = null;
+			if (file == null) {
+				file = new RandomAccessFile(path.toFile(), "rw");
+			}
+			try (FileChannel channel = file.getChannel()) {
+
+				try (FileLock lock = channel.tryLock(channel.size(),
+						channel.size() + text.length() + System.lineSeparator().length(), true)) {
+					if (lock != null) {
+						String text = "Hello, world.";
+						buffer = ByteBuffer.allocate(text.length() + System.lineSeparator().length());
+						buffer.put((text + System.lineSeparator()).getBytes(Charsets.UTF_8));
+						buffer.flip();
+						while (buffer.hasRemaining()) {
+							channel.write(buffer, channel.size());
+						}
+					}
+				} catch (OverlappingFileLockException e) {
+					// Failed to lock. Try again later.
+					Thread.sleep(50);
+					lockAndWrite();
+				}
+			} catch (IOException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+
+		}
+
+	}
+
+	public static void main(String[] args) throws InterruptedException, IOException {
+		Path path = Paths.get("/tmp/foo");
+		Files.deleteIfExists(path);
+		Files.createFile(path);
+		int concurrentWriters = 5;
+		CountDownLatch countDownLatch = new CountDownLatch(concurrentWriters);
+		// Launch 10 writers in parallel
+		final AtomicInteger count = new AtomicInteger(0);
+		Stream.generate(() -> new Thread(new Writer(path, "foo " + count.incrementAndGet(), countDownLatch)))
+				.limit(concurrentWriters).forEach(Thread::start);
+
+		countDownLatch.await();
+		AtomicInteger lineCount = new AtomicInteger(0);
+		Files.lines(path).forEach((line) -> {
+			lineCount.incrementAndGet();
+			System.out.println(line);
+		});
+		log.info("Total lines written = " + lineCount.get());
```

Comment:
With slf4j we don't need to use the + to concat we can use for example:
logger.debug("Printing variable value: {}", variable);

Which is a bit more elegant.

Reply:
This code was deleted too, for the same reason.